### ISSUE TITLE
Dropped Chameleon support?

### ISSUE THREAD

**Issue by mikeckennedy** [Pylons/pyramid-cookiecutter-starter#12]
Hi guys,

Looks like you dropped chameleon support for the old starter scaffold.

I created a variant here:

https://github.com/mikeckennedy/pyramid-cookiecutter-starter-chameleon

I love that the new cookie cutter comes with a layout. I might add the layout to this template. But in general would you be willing to take it back into this organization?

**Comment by stevepiercy** [comment]
For historical perspective, the switch from Chameleon to Jinja2 for the starter cookiecutter was made because Jinja2 is the more popular and well-known templating language for Python developers, and our available volunteer time in the Pylons Project was insufficient to maintain additional official cookiecutters.

I'd propose the following steps be fulfilled, which would satisfy my concerns, to bring it back under the Pylons Project organization:

1. No licensing issues (consistent with http://pylonsproject.org/community-how-to-contribute.html and http://pylonsproject.org/license.html)
1. @mikeckennedy's repo is satisfactory to bring under the Pylons Project GitHub organization (@stevepiercy volunteers to review code)
1. @mikeckennedy would take the lead to maintain this cookiecutter
1. @mikeckennedy would be added to the cookiecutter group of the Pylons Project GitHub organization

Ping @mmerickel and @bertjwregeer for discussion.

**Comment by mmerickel** [comment]
It might be worth considering just adding the template engine as a choice in `cookiecutter.json` and then adding logic to the various templates to render things appropriately for either option. I'm somewhat unenthused about adding a 4th "official" scaffold as the maintenance required on them is quite a lot (we try hard to keep the official ones in sync with each other). Another option would be to establish somewhere known third-party pyramid-based cookiecutters. This is just me thinking out loud.. don't take it as a decision of any sort.

Adding a layout example with a lot would be wise I think, it's the most common question I've seen from any newcomer asking about chameleon.

**Comment by stevepiercy** [comment]
I think I should clarify what I meant by "official". There are three official Pyramid cookiecutters, and I do not propose to change that. I meant official under the Pylons Project organization, but not officially supported by Pyramid and its core developers. I meant it would be like the [SubstanceD cookiecutter](https://github.com/pylons/substanced-cookiecutter) that is supported by @cguardia.

> Another option would be to establish somewhere known third-party pyramid-based cookiecutters.

This option exists now in two places.
* https://github.com/audreyr/cookiecutter/#python-pyramid
* https://trypyramid.com/resources-extending-pyramid.html

**Comment by mikeckennedy** [comment]
Hi guys,

@stevepiercy  I think I would be up for supporting the template. But as @mmerickel pointed out it might be nice to just add chameleon support to the existing one. It's really just like 3 lines of code / config + swapping out the mytemplate.jinja2 for mytemplate.pt. So I would prefer to see that as the solution to simplify it for the community.

It's worth pointing out why I started down this path yesterday. I recorded my class using chameleon and now people who are starting the class have literally no scaffold to create a chameleon based app. I'm opening a separate issue so we can keep these discussion separate: 

Lucky #13 **pcreate -s starter app_name is not chameleon**

Thanks,
Michael

**Comment by stevepiercy** [comment]
I'd go along with adding Chameleon as an option to the starter cookiecutter, keeping Jinja2 as the other option. There would be minor docs updates, too, but I can handle those if you handle the cookiecutter code. There was also an [earlier version of the starter cookiecutter that used Chameleon](https://github.com/Pylons/pyramid-cookiecutter-starter/commit/fbc706833053e427cf6b6139455902fcdba165a7), so you might be able to recycle that code and template, making sure that subsequent tweaks to the template got merged in as well.

Would you submit a PR with the modifications?

**Comment by mikeckennedy** [comment]
Sure, I can do it. Just have to learn some Cookie Cutter and I'll get it done. :)

**Comment by mmerickel** [comment]
> I think I should clarify what I meant by "official".

As far as adding the repo to the Pylons org, I have no problem with that at all.

**Comment by stevepiercy** [comment]
@mikeckennedy we got a potential solution!

Question:
https://gitter.im/audreyr/cookiecutter?at=5893cc792ff29dec066e22f7 

Answer:
https://gitter.im/audreyr/cookiecutter?at=5895e65dca6c6eb44945896f

Refs:
https://github.com/pydanny/cookiecutter-django/blob/master/hooks/post_gen_project.py

**Comment by mikeckennedy** [comment]
Nice. I think that's the best solution. How is this moving forward? Is one of you making the change or are you waiting on me to learn cookie cutter? ;)

**Comment by mmerickel** [comment]
> or are you waiting on me to learn cookie cutter?

Yes please.

**Comment by stevepiercy** [comment]
What @mmerickel said. I'm up to my eyeballs ATM.

**Comment by mikeckennedy** [comment]
Hi guys. I think I fixed it. Have a look. Not sure if the CI needs to be updated due to one more choice in the cookiecutter process.

https://github.com/Pylons/pyramid-cookiecutter-starter/pull/14

**Comment by mikeckennedy** [comment]
This is fixed, thanks for the support. When will it appear in pcreate?

**Comment by mmerickel** [comment]
Ah pcreate is a separate issue. I'm happy to consider any PRs in that area (including for 1.8.x) but as it's deprecated in favor of the cookiecutters I don't plan on porting any changes over myself.

**Comment by mmerickel** [comment]
> Not sure if the CI needs to be updated due to one more choice in the cookiecutter process.

This is a good point, I can look at this tonight as well if no one else has time. We should ensure the tests are running for both cookiecutter options. This should be pretty straightforward.

**Comment by stevepiercy** [comment]
It's all you, @mmerickel. I'm overbooked this week.

**Comment by mikeckennedy** [comment]
Ok, thanks guys.
